mkosi/__init__.py

```diff
@@ -3229,10 +3227,8 @@ def run_finalize_script(config: MkosiConfig, state: MkosiState, for_cache: bool)
     verb = "build" if state.do_run_build_script else "final"
 
     with complete_step("Running finalize script…"):
-        env = collections.ChainMap(dict(BUILDROOT=str(state.root), OUTPUTDIR=str(output_dir(config))),
-                                   config.environment,
-                                   os.environ)
-        run([config.finalize_script, verb], env=env)
+        run([config.finalize_script, verb],
+            env={**config.environment, "BUILDROOT": str(state.root), "OUTPUTDIR": str(output_dir(config))})
```

Comment:
I just noticed that `os.environ` is missing from here. Is that intentional?

Reply:
Yes it's added automatically by run()